scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/license/LicenseSchedulingPolicy.java

```diff
@@ -171,26 +192,29 @@ private boolean acquireTokensForJob(JobDescriptorImpl job) {
 
             // Do not execute if one of the required license can not be obtained
             final HashSet<String> requiredLicensesSet = new HashSet<String>(Arrays.asList(requiredLicenses.split("\\s*,\\s*")));
+            boolean jobHoldsToken, canGetToken;
             for (String requiredLicense : requiredLicensesSet) {
 
                 if (!licenseSynchronization.containsLicense(requiredLicense)) {
                     logger.debug("The required software is not specified in the license properties file, keep job pending");
                     return false;
                 }
 
-                if (!licenseSynchronization.containsJobId(requiredLicense, jobId) && !canGetToken(requiredLicense)) {
-                    logger.debug("Token for " + requiredLicense + " not available, keep job pending");
+                jobHoldsToken = licenseSynchronization.containsJobId(requiredLicense, jobId);
+                canGetToken = canGetToken(requiredLicense);
+                if (!jobHoldsToken && !canGetToken) {
+                    logger.debug("Job does not already hold a token for " + requiredLicense +
+                                 " and cannot get a new one, keep job pending");
```

Comment:
good improvement, if a job does not already hold or can acquire a token for all required licences, then you immediately return false without acquiring tokens partially

Reply:
This is a remark or a new condition to consider ?